File: Chat.Api/ChatService/Commands/SendMessageCommand.cs
using Chat.Api.ChatService.Interfaces;
using Chat.Api.Core.Models;
using Chat.Api.Core.Services;
using Chat.Api.ChatService.Models;
using Chat.Api.Core.Interfaces;

namespace Chat.Api.ChatService.Commands
{
    public class SendMessageCommand : ACommand
    {
        public ChatModel ChatModel {get; set;}
        
        public override void ValidateCommand()
        {
            if (ChatModel == null) 
            {
                throw new Exception("ChatModel not set");
            }
        }
    }
}

File: chat-app/src/app/chat/commands/send-message-command.ts
import { CommandBase } from "src/app/core/models/command-base";
import { Configuration } from "src/app/core/services/configuration";
import { ChatModel } from "../models/chat-model";

export class SendMessageCommand extends CommandBase {
    
    chatModel : ChatModel;
    
    constructor() {
        super();
        this.chatModel = new ChatModel();
        this.apiUrl = Configuration.identityApi + "/chat/send"
    }
}